adapters/InMobi/InMobiAdapter/GADMAdapterInMobiInterstitialAd.m

```diff
@@ -0,0 +1,180 @@
+        return delegate;
+    };
+    
+    GADMAdapterInMobiInterstitialAd *__weak weakSelf = self;
+    NSString *accountID = _interstitialAdConfig.credentials.settings[GADMAdapterInMobiAccountID];
+    [GADMAdapterInMobiInitializer.sharedInstance
+     initializeWithAccountID:accountID
+     completionHandler:^(NSError *_Nullable error) {
+        GADMAdapterInMobiInterstitialAd *strongSelf = weakSelf;
+        if (!strongSelf) {
+            return;
+        }
+        
+        if (error) {
+            NSLog(@"[InMobi] Initialization failed: %@", error.localizedDescription);
+            strongSelf->_interstitialRenderCompletionHandler(nil, error);
+            return;
+        }
+        
+        [strongSelf requestInterstitialAd];
+    }];
+}
+
+- (void)requestInterstitialAd {
+    long long placementId =
+    [_interstitialAdConfig.credentials.settings[GADMAdapterInMobiPlacementID] longLongValue];
+    if (placementId == 0) {
+        NSError *error = GADMAdapterInMobiErrorWithCodeAndDescription(
+                                                                      GADMAdapterInMobiErrorInvalidServerParameters,
+                                                                      @"[InMobi] Error - Placement ID not specified.");
+        _interstitialRenderCompletionHandler(nil,error);
+        return;
+    }
+    
+    if ([_interstitialAdConfig isTestRequest]) {
+        NSLog(@"[InMobi] Please enter your device ID in the InMobi console to receive test ads from "
+              @"InMobi");
+    }
+    
+    _interstitialAd = [[IMInterstitial alloc] initWithPlacementId:placementId];
+    
+    GADInMobiExtras *extras =  _interstitialAdConfig.extras;
+    if (extras && extras.keywords) {
+        [_interstitialAd setKeywords:extras.keywords];
+    }
+    
+    GADMAdapterInMobiSetTargetingFromAdConfiguration(_interstitialAdConfig);
+    NSDictionary<NSString *, id> *requestParameters =
+    GADMAdapterInMobiCreateRequestParametersFromAdConfiguration(_interstitialAdConfig);
+    [_interstitialAd setExtras:requestParameters];
+    
+    _interstitialAd.delegate = self;
+    [_interstitialAd load];
+}
+
+- (void)presentFromViewController:(nonnull UIViewController *)viewController {
+    if ([_interstitialAd isReady]) {
+        [_interstitialAd showFromViewController:viewController
+                                  withAnimation:kIMInterstitialAnimationTypeCoverVertical];
+    }  else {
+        NSError *error = GADMAdapterInMobiErrorWithCodeAndDescription(
+            GADMAdapterInMobiErrorAdNotReady,
+            @"[InMobi] Error - Interstitial ad not ready to be present.");
+        [_interstitalAdEventDelegate didFailToPresentWithError:error];
+    }
+}
+
+- (void)stopBeingDelegate {
+    _interstitialAd.delegate = nil;
+}
+
+#pragma mark IMAdInterstitialDelegate methods
+
+- (void)interstitialDidFinishLoading:(nonnull IMInterstitial *)interstitial {
+    NSLog(@"<<<< interstitialDidFinishRequest >>>>");
```

Comment:
See comments for banner ads above.

The same applies to all the other callbacks.

Reply:
done